src/java/org/apache/cassandra/db/memtable/Flushing.java

```diff
@@ -299,16 +299,13 @@ public static SSTableMultiWriter createFlushWriter(ColumnFamilyStore cfs,
                                                        Descriptor descriptor,
                                                        long partitionCount)
     {
-        MetadataCollector sstableMetadataCollector = new MetadataCollector(flushSet.metadata().comparator)
-                                                     .commitLogIntervals(new IntervalSet<>(flushSet.commitLogLowerBound(),
-                                                                                           flushSet.commitLogUpperBound()));
-
         return cfs.createSSTableMultiWriter(descriptor,
                                             partitionCount,
                                             ActiveRepairService.UNREPAIRED_SSTABLE,
                                             ActiveRepairService.NO_PENDING_REPAIR,
                                             false,
-                                            sstableMetadataCollector,
+                                            new IntervalSet<>(flushSet.commitLogLowerBound(),
```

Comment:
If I understand correctly, this is the main change. Instead of passing a shared collector we pass positions and later on create a collector per sstable. Should we add a test for this?

Reply:
It is no longer possible to pass a collector thus the same problem can't appear, but I added a test that the cardinality is correct after flushing with multiple resulting sstables (failing without the patch).